edx_rest_framework_extensions/permissions.py

```diff
@@ -1,9 +1,71 @@
 """ Permission classes. """
+from oauth2_provider.ext.rest_framework.permissions import TokenHasScope
```

Comment:
I think that is okay and should be expected given that this is just a customization of what the DOT permission class does.

Reply:
Right.  Although I was hoping that all Authentication Provider implementation can remain in a single Authentication service - rather than needing to be updated/maintained in all IDAs.  Anyway, I don't see an alternative unless we don't inherit from that class - so it's a tradeoff, which is fine for now.

Reply:
Fair point. I'll just copy over the [get_scopes](https://github.com/jazzband/django-oauth-toolkit/blob/master/oauth2_provider/contrib/rest_framework/permissions.py#L54) function from `TokenHasScope` and inherit from `BasePermission`.